### ISSUE TITLE
AllowedValues linting for DMS::Endpoint has a typo

### ISSUE THREAD

**Issue by alytle** [aws-cloudformation/cfn-python-lint#774]
*cfn-lint version: cfn-lint 0.17.0

*Description of issue.*

Adding an AllowedValues condition to a Parameter used for a DMS::ReplicationEndpoint has a typo which causes a false warning:

```
W2030 You must specify a valid allowed value for SourceEngine (mariadb).
Valid values are ['aurora-postgresql', 'aurora', 'azuredb', 'db2', 'dynamodb', 'mariad', 'mongodb', 'mysql', 'oracle', 'postgres', 'redshift', 's3', 'sqlserver', 'sybase']
dms.yaml:36:9
```

Note the 'mariad' entry should be 'mariadb'.

The Resource Specification and Documentation appear to be correct:

```
EngineName

    The type of engine for the endpoint. Valid values depend on the EndPointType and include mysql, oracle, postgres, **mariadb**, aurora, aurora-postgresql, redshift, s3, db2, azuredb, sybase, dynamodb, mongodb, and sqlserver. 
```

**Comment by alytle** [comment]
Created PR for the typo: https://github.com/aws-cloudformation/cfn-python-lint/pull/775

**Comment by kddejong** [comment]
Please try release v0.17.1
